webpack.config.js

```diff
@@ -153,17 +175,7 @@ const config = {
       ".jade",
       ".coffee",
     ],
-    modules: [
-      path.resolve(__dirname, "src"),
-      "node_modules",
-      // TODO: The problem with including /all/ node_modules directories is that
-      //       when using npm-link it can bring in dev dependencies that we
-      //       don’t need or worse are duplicates and lead to bugs. However,
-      //       ignoring all non-hoisted modules now would probably lead to bugs
-      //       as well.
-      // path.resolve(__dirname, "node_modules"),
-      // path.resolve(__dirname, "node_modules/@artsy/reaction/node_modules"),
-    ],
+    modules: [path.resolve(__dirname, "src"), "node_modules"],
```

Comment:
When you link reaction, there’s a big chance it will have its own `node_modules` dir with dependencies that it would not have had if reaction had not been linked. Instead no dev dependencies of reaction would have been installed and where possible packages would have been hoisted up to force’s node_modules.

The problem now is that reaction may be using unexpected code. One I’ve often dealt with in the past was that graphql-js would use instanceof on an object which would have the same class name, but actually be different classes. These are not fun bugs to deal with (luckily they don’t happen often).

I bet that’s what these last commenters are probably dealing with too https://github.com/webpack/webpack/issues/811

Reply:
This makes sense, I wonder what a solution would be (but i guess that is for the TODO 😄).

Reply:
Yuuuuup :D